src/django/api/views/v1/production_locations/production_locations.py

```diff
@@ -0,0 +1,24 @@
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from api.services.search import OpenSearchService
+from api.views.v1.opensearch_query_builder \
+    import OpenSearchQueryDirector, OpenSearchQueryBuilder
+
+
+class ProductionLocations(APIView):
```

Comment:
1. `ViewSet` is now using instead of `APIView` for ProductionLocations class.
2. I created handlers for client and server errors. For 500 error this is specifically an error about losing connection to the OpenSearch. For client error:  error triggers for improper setting of `number_of_workers`, `percent_female_workers` and `coordinates`. If we define rules for other parameters, we can expand param serializer class (aka `ProductionLocationsSerializer`) that I've introduced in this PR.